social_panel/lib/twitter_client.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require 'faraday'
+require 'JSON'
+require 'pry'
+
+class TwitterClient
+
+  # URL = 'https://api.twitter.com/1.1'
+  # TOKEN = "Bearer #{ENV.fetch('SOCIAL_PANEL_TWITTER_ACCESS_TOKEN')}"
+
+  def connection
+      Faraday.new(
+        url: 'https://api.twitter.com/1.1',
+        headers: { 'Authorization' => "Bearer #{ENV['SOCIAL_PANEL_TWITTER_ACCESS_TOKEN']}" },
+      )
+  end
```

Comment:
Really nice idea @ikantoch 🥇

Good job.

Reply:
There is one interesting implication of this. For example when I run this code:

```ruby
get_tweet_by_id(1)
get_tweet_by_id(2)
get_tweet_by_id(3)
```

the `Faraday.new` will be executed 3 times and we can think about it - is it good or not, etc.

We will talk about it tomorrow :)